src/client.js

```diff
@@ -60,3 +61,27 @@ function refresh_token (_refresh_token) {
             return result;
         });
 }
+
+function revoke_token (token, client_id, type = '') {
+    const scopes_string = cache.find_key('client', {token});
+    const payload = {
+        client_id: client_id,
+        type: type,
+    };
+    const access_token = `Bearer ${token}`;
+
+    return cudl.post
+        .to(config.base_url + config.path + '/oauth/revoke')
+        .set_header('Authorization', access_token)
+        .send(payload)
+        .max_retry(config.retry_count)
+        .promise()
+        .then(result => {
+            if (scopes_string) {
+                cache.forget('client', token);
+                cache.forget('server', token);
```

Comment:
Why do we need to also remove it from the 'server'? I think we should do the remove based on if client_id is provided?

Reply:
On verify scope, this data is also saved in `server` that is why we need to remove it as well.

Reply:
This is now already updated. I have included the key on server cache as well.